src/System.Dynamic.Runtime/src/System.Dynamic.Runtime.csproj

```diff
@@ -9,7 +9,7 @@
     <AssemblyName>System.Dynamic.Runtime</AssemblyName>
     <RootNamespace>System.Dynamic.Runtime</RootNamespace>
     <AssemblyVersion>4.0.11.0</AssemblyVersion>
-    <DefineConstants Condition=" '$(IsInterpreting)' != 'true' ">$(DefineConstants);FEATURE_CORECLR</DefineConstants>
+    <DefineConstants Condition=" '$(IsInterpreting)' != 'true' ">$(DefineConstants);FEATURE_COMPILER</DefineConstants>
```

Comment:
FEATURE_COMPILER => FEATURE_COMPILE

This looks like the root cause for issue https://github.com/dotnet/corefx/pull/4862. While that issue was fixed by removing the #ifdef it looks like we might also have some other bugs in this library. @DinoV @VSadov @bartdesmet I'm going to file a separate issue to fix this.